src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -2734,8 +2735,12 @@ kernel_arg_transaction_execute (RpmostreedTransaction *transaction,
       for (char **iter = self->kernel_args_deleted; iter && *iter; iter++)
         {
           const char*  arg =  *iter;
-          if (!ostree_kernel_args_delete (kargs, arg, error))
-            return FALSE;
+          if (g_strv_contains (existing_kargs, arg))
+            {
+              changed = TRUE;
+	      if (!ostree_kernel_args_delete (kargs, arg, error))
```

Comment:
This function already errors out if the karg doesn't exist (this is why we have to use the `existing_kargs` hack for the new idempotent switches and what https://github.com/ostreedev/ostree/issues/2329 is about).

Since that's the historical behaviour, we should maintain it and keep erroring out if the karg is missing.